src/Traitor.jl

```diff
@@ -54,56 +54,68 @@ previous definition.
 """
 module Traitor
 
-import Base: @pure
-export @traitor, @betray, supertrait
+using Base: @pure, uncompressed_ast, unwrap_unionall, tuple_type_cons
+using Core: SimpleVector, svec, CodeInfo
 
+export @traitor, supertrait, betray!
 
 """
-    @betray f
+    betray!(f, ::Type{TT}) where {TT <: Tuple}
 
-The goal of the `@betray` macro is to take over the methods of `f` and ready
-them to be compatible with traits-based dispatch, so that new submethods
-defined by `@traitor` do not overwrite existing default methods (and used where
-no more specific trait match is found).
+Et tu?
 
-WARNING: Please note this macro has NOT been implemented yet! (If anyone knows
-how to turn a `Method` object into a new method definition, please let us know).
+Take over the methods of `f` and readythem to be compatible with traits-based dispatch, 
+so that new submethods defined by `@traitor` do not overwrite existing default methods 
+(and used where no more specific trait match is found).
 
-(See also `Traitor` and `@traitor`.)
+Betraying functions is not a very kind thing to do, and can cause unintended side effects.
```

Comment:
Do you have examples of these unintended side effects?

Reply:
Hm. Well, one example would be if you `betray!`'d `Traitor.get_trait_table`, you could end up deleting the trait methods for a given signature.